scipy/optimize/tests/test_slsqp.py

```diff
@@ -564,3 +586,22 @@ def test_gh9640(self):
 
         # The problem is infeasible, so it cannot succeed
         assert not res.success
+
+    def test_parameters_stay_within_bounds(self):
+        # gh11403. For some problems the SLSQP Fortran code suggests a step
+        # outside one of the lower/upper bounds. When this happens
+        # approx_derivative complains because it's being asked to evaluate
+        # a gradient outside its domain.
+        np.random.seed(1)
+        bounds = Bounds(np.array([0.1]), np.array([1.0]))
+        n_inputs = len(bounds.lb)
+        x0 = np.array(bounds.lb + (bounds.ub - bounds.lb) *
+                      np.random.random(n_inputs))
+
+        def f(x):
+            assert (x >= bounds.lb).all()
+            return np.linalg.norm(x)
+
+        with pytest.warns(RuntimeWarning):
```

Comment:
More explicit and specific
```suggestion
        with pytest.warns(RuntimeWarning, match='were outside bounds'):
```

Reply:
Ack